### PR TITLE
Fix memory leaks

### PR DESCRIPTION
**Pull request by ghost**
### Description

Fixed removing of listeners in some methods with `once` listenerswhere only one of them could fire (except case of timeout, when no one is triggered and removed), and the second remained in memory 

Tested using the benchmark system with the following changes: https://github.com/bitfinexcom/bfx-hf-algo/pull/158

#### Before:

image

#### After

image

### Fixes:
- [x] Memory leak on order sumit
- [x] Minor leaks on order update/cancel and auth

### PR status:
- [x] Tests added or updated

### CODE REVIEW

File: lib/ws2/auth.js [hunk 1]
```diff
@@ -19,22 +19,13 @@ const send = require('./send')
...
-    ev.once('event:auth:error', err => {
-      debug('error authenticating: %j', err)
-      reject(err)
-    })
+  send(state, authParams)
 
-    send(state, authParams)
-  })
+  return watchResponse(ev, 'event:auth:success', 'event:auth:error')
+    .then(() => { debug('authenticated') })
```
Comment (avsek477) [inline lib/ws2/auth.js:27]: this will always return auth operation as a success and will never reject on auth error

File: lib/ws2/auth.js [hunk 2]
```diff
@@ -19,22 +19,13 @@ const send = require('./send')
...
+    .catch(err => { debug('error authenticating: %j', err) })
```
Comment (avsek477) [inline lib/ws2/auth.js]: This will always return auth operation as a success and will not throw auth error. Should throw an error in catch block.

Comment (ghost) [reply lib/ws2/auth.js]: good catch! Forgot about reject() in the `event:auth:error` hanler

**Review by avsek477 — APPROVED** [review]
lgtm

**Comment by robertkowalski** [comment]
nice catch!

**Review by robertkowalski — APPROVED** [review]
lgtm

**Pull request merged**